### PR TITLE
Added UI for upload

### PR DESCRIPTION
**Pull request by DivergentEuropeans**
image

Multi-file upload doesn't work yet, but the building blocks are there. We need a queue, perhaps

Signed-off-by: Leanid Astrakou <lastrakou@rocketsoftware.com>

### CODE REVIEW

**Comment by DivergentEuropeans** [comment]
Needs changelog

**Comment by DivergentEuropeans** [comment]
This is also missing file encoding info in the modal

File: src/app/services/uploader.service.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+*/
+
+import { Inject, Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class UploaderService {
+    constructor(private http: HttpClient, 
+        @Inject(Angular2InjectionTokens.LOGGER) private log: ZLUX.ComponentLogger) { }
```
Comment (lchudinov) [inline src/app/services/uploader.service.ts]: import for `Angular2InjectionTokens` is missing at the top of the file.

File: src/app/services/uploader.service.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                const chunkSize =  3 * 1024 * 1024; // bytes
+                let sourceEncoding;
+                targetEncoding === 'BINARY' ? sourceEncoding = 'BINARY' : sourceEncoding = 'UTF-8';
+                let chunkIdx = 0;
+                let offset = 0;
+                let sessionID: number;
+                const uri = ZoweZLUX.uriBroker.unixFileUri('contents', uploadDirPath.slice(1) + '/' + file.name,
+                sourceEncoding, targetEncoding, undefined, true);
+
+                console.table({'URI': uri, 'File Name': file.name, 'File Size': fileSize, 'Chunk Size': chunkSize});
```
Comment (lchudinov) [inline src/app/services/uploader.service.ts]: `console.table`

Comment (lchudinov) [inline src/app/services/uploader.service.ts]: `console.table`

**Comment by lchudinov** [comment]
Here is what I see:
image

* File uploading works in case of a single file
* The encoding field should have a label describing its purpose
* When the file has uploaded I have to manually refresh the folder to see the new file in the tree. It would be convenient if the tree updated automatically.

File: src/app/components/upload-files-modal/upload-files-modal.component.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            <a href="https://www.ibm.com/docs/en/zos/2.1.0?topic=ccsids-encoding-scheme" target="_blank">
+              <i class="fa fa-question-circle dataset-properties-question-circle"></i>
+            </a>
+        </li>
+    </ul>
+</div>
+
+<mat-dialog-actions>
+  <button mat-button mat-dialog-close class="modal-mat-button cancel">Cancel</button>
+  <button mat-button mat-dialog-close class="modal-mat-button create" (click)="uploadHandlerSetup()">Upload</button>
```
Comment (lchudinov) [inline src/app/components/upload-files-modal/upload-files-modal.component.html]: Currently, I can click the Upload even if I haven't selected any file or not selected an encoding. Please add `[disabled]` attribute with an appropriate condition in order to enable the button only when the dialog is filled in correctly.

**Comment by lchudinov** [comment]
The dialog looks much better, but the file tree still doesn't update automatically when I've upload a file.

**Comment by lchudinov** [comment]
now it updates the tree after the file upload.

File: src/app/components/upload-files-modal/upload-files-modal.component.html (hunk 3 above)
Comment (lchudinov) [reply src/app/components/upload-files-modal/upload-files-modal.component.html]: it's fixed.

**Review by lchudinov — APPROVED** [review]

**Pull request merged**